gli/utils.py

```diff
@@ -392,3 +392,366 @@ def save_data(prefix, **kwargs):
     for key, matrix in sparse_arrays.items():
         sp.save_npz(f"{prefix}_{key}.sparse.npz", matrix)
         print("Save sparse matrix", key, "to", f"{prefix}_{key}.sparse.npz")
+
+class HjsonTemplate(object):
+    def __init__(
+        self,
+        name
+    ):
+        self.missing_features_messages_error = []
+        self.missing_features_messages_warning = []
+        self._name = name
+
+    def _check_file_and_key(self, d, v, k):
+        if "file" not in d:
+            self._missing_features_messages_warning.append("%s missing 'file' in %s" % (self._name, v))
+            d.update({"file" : "%s_task.npz" % self._name})
+        if "key" not in d:
+            self._missing_features_messages_warning.append("%s missing 'key' in %s" % (self._name, v))
+            d.update({"key" : k})
+        return d
+    
+    def metadata(
+        self,
+        node=dict(),
+        edge=dict(),
+        graph=dict(),
+        citation=None,
+        is_heterogeneous=True
+    ):
+        # missing_features_messages_error = []
+        # missing_features_messages_warning = []
+        self._node  = node
+        if "_Edge" in edge:
+            edge.update({"_Edge" : self._check_file_and_key(edge.get("_Edge"), "_Edge", "edge")})
+        else:
+            self._missing_features_messages_error.append("%s missing '_Edge'" % self._name) 
+        self._edge = edge
+        if "_NodeList" in graph:
+            if "file" not in graph.get("_NodeList"):
+                self._missing_features_messages_warning.append("%s missing 'file' in '_NodeList'" % self._name)
+                graph.update({"file" : "%s.npz" % self._name})
+            if "key" not in graph.get("_NodeList"):
+                self._missing_features_messages_warning.append("%s missing 'key' in '_NodeList'" % self._name)
+                graph.update({"key" : "node_list"})
+        else:
+            self._missing_features_messages_error.append("%s missing '_NodeList'" % self._name)
+        if "_EdgeList" in graph:
+            if "file" not in graph.get("_EdgeList"):
+                self._missing_features_messages_warning.append("%s missing 'file' in '_EdgeList'" % self._name)
+                graph.update({"file" : "%s.npz" % self._name})
+            if "key" not in graph.get("_EdgeList"):
+                self._missing_features_messages_warning.append("%s missing 'key' in '_EdgeList'" % self._name)
+                graph.update({"key" : "edge_list"})
+        self._graph = graph 
+        self._citation = citation
+        self._is_heterogeneous = is_heterogeneous
+
+        # return missing_features_messages_error, missing_features_messages_warning
+    
+    def save_metadata(self):
```

Comment:
Perhaps we can directly save the data when user calls the `metadata` function?
What prevents us from doing so?